src/web/components/App.tsx

```diff
@@ -429,6 +420,186 @@ export const App = ({ CAPI, NAV, ophanRecord }: Props) => {
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/CalloutBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.CalloutBlockComponent,
+		},
+	);
+
+	const DocumentBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.DocumentBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/DocumentBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.DocumentBlockComponent,
+		},
+	);
+
+	const EmbedBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.EmbedBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/EmbedBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.EmbedBlockComponent,
+		},
+	);
+
+	const UnsafeEmbedBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.EmbedBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/UnsafeEmbedBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.UnsafeEmbedBlockComponent,
+		},
+	);
+
+	const InstagramBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.InstagramBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/InstagramBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.InstagramBlockComponent,
+		},
+	);
+
+	const MapEmbedBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.MapBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/MapEmbedBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.MapEmbedBlockComponent,
+		},
+	);
+
+	const SpotifyBlockComponent = loadable(
+		() => {
+			if (
+				CAPI.elementsToHydrate.filter(
+					(element) =>
+						element._type ===
+						'model.dotcomrendering.pageElements.SpotifyBlockElement',
+				).length > 0
+			) {
+				return import(
+					'@frontend/web/components/elements/SpotifyBlockComponent'
+				);
+			}
+			return Promise.reject();
+		},
+		{
+			resolveComponent: (module) => module.SpotifyBlockComponent,
+		},
+	);
```

Comment:
I wish that these loadable function callbacks could be less repetitive, but if I remember correctly from my original implementation it messes with types when doing that.